packages/loopring_v3/contracts/aux/access/LoopringIOExchangeOwner.sol

```diff
@@ -187,6 +209,23 @@ contract LoopringIOExchangeOwner is SelectorBasedAccessManager, ERC1271, Drainab
         }
     }
 
+    function _afterBlockSubmission(
+        ExchangeData.Block[] memory   /*blocks*/,
+        PostBlocksCallback[] calldata postBlocksCallbacks
+        )
+        private
+    {
+        for (uint i = 0; i < postBlocksCallbacks.length; i++) {
+            // Disallow calls to the exchange and pre block callback contracts
+            require(postBlocksCallbacks[i].to != target, "EXCHANGE_CANNOT_BE_POST_CALLBACK_TARGET");
+            require(postBlocksCallbacks[i].data.toBytes4(0) != IBlockReceiver.beforeBlockSubmission.selector, "INVALID_POST_CALLBACK_FUNCTION");
```

Comment:
Did it like this because we know exactly which function cannot be called, and it's much cheaper than having to check if an address is an agent or not.

Reply:
I wonder if someone can use an global agent as `to` to attach us or a user.

Reply:
If a global agent allows doing stuff for users it will have it's own authorization method. If the authorization method is simply `msg.sender = exchange.owner` there could be a problem (like for the ITransactionReceivers is the case), but otherwise I don't see why the owner would have any special powers in other contracts (except the exchange itself of course).